### PR TITLE
Install PyTorch with pip

### LINKED ISSUE #67: speed up CI tests (do we need all conda packages?)
By using Conda, unfortunately the setup phase of the CI jobs have become a bit slower than without conda, maybe we can look into speeding them up again by checking whether we need all the packages that we install during CI

### PR DESCRIPTION
**Pull request by matthewfeickert**
Now that [PyTorch is on PyPI](https://twitter.com/pytorch/status/981198007352705025) Conda is no longer required to get an easy install of stable builds. As PyTorch was the only reason Conda was being installed during CI Miniconda no longer needs to be part of the Travis YAML file. This should result in a speedup of the CI.

Resolves #67 

# Checklist Before Requesting Approver

- [x] Tests are passing
- [x] "WIP" removed from the title of the pull request

### CODE REVIEW

**Comment by matthewfeickert** [comment]
Still getting the memory error after all tests pass:
```
=================== 24 passed, 26 warnings in 156.84 seconds ===================
*** Error in `/home/travis/virtualenv/python3.6.3/bin/python': double free or corruption (!prev): 0x000000000137f0f0 ***
/home/travis/.travis/job_stages: line 57:  4879 Aborted                 (core dumped) pytest --ignore tests/benchmarks/
```
So further work needs to be done.

**Comment by matthewfeickert** [comment]
Using the [Python TravisCI Docker testing image](https://docs.travis-ci.com/user/common-build-problems/#Troubleshooting-Locally-in-a-Docker-Image) I was able to see that the reason that in the past the builds have succeeded on the Python 3.6 tests in Travis but failed in Python 2 is that the Travis Python 2 version is too old.

```
$ pip install -U --process-dependency-links -e .[develop]
...
Collecting tornado>=4.0 (from ipykernel->jupyter->pyhf==0.0.8)
  Using cached https://files.pythonhosted.org/packages/cf/d1/3be271ae5eba9fb59df63c9891fdc7d8044b999e8ac145994cdbfd2ae66a/tornado-5.0.2.tar.gz
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-build-Wk7Aqw/tornado/setup.py", line 146, in <module>
        raise ImportError("Tornado requires an up-to-date SSL module. This means "
    ImportError: Tornado requires an up-to-date SSL module. This means Python 2.7.9+ or 3.4+ (although some distributions have backported the necessary changes to older versions).
    
    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-build-Wk7Aqw/tornado/
You are using pip version 9.0.1, however version 10.0.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
$ python --version
Python 2.7.6
```

**Comment by kratsg** [comment]
Can you report this to Travis-CI? I don't see an existing issue for it. This seems like a weird problem though.

**Comment by matthewfeickert** [comment]
Having to force upgrade `numpy` after everything else has been installed feels like a really dumb patch to get this everything to build. I'd like to understand what is going on better before this gets finished up.

**Comment by matthewfeickert** [comment]
@lukasheinrich @kratsg  I think this is ready for review. push CI builds now run in under 6 minutes, and pull CI builds should run in around 10 minutes.

If there are no comments or change requests please squash and merge this in.

**Comment by lukasheinrich** [comment]
Hi @matthewfeickert can you remove the `--process-dependency-links` in the setup.py? I think I carried this over but it's actually not a recommended practice. Also we don't have any custom dependency_links anyways.

**Comment by matthewfeickert** [comment]
@lukasheinrich:
> can you remove the `--process-dependency-links` in the `setup.py`>

Removed in 5904994

File: .travis.yml [hunk 1]
```diff
@@ -4,25 +4,19 @@ python:
...
-  - conda info -a
-  - conda create -q -n test-environment python=$TRAVIS_PYTHON_VERSION pytorch scipy -c pytorch
-  - source activate test-environment
-  - pip install -U --process-dependency-links -e .[develop]
-  - pip install --upgrade pytest
+- sudo apt-get update
+- pip install --upgrade pip
+- pip install -U -q -e .[develop]
+# numpy upgrade needed for Python 2.7 only
+- pip install --upgrade pytest numpy
```
Comment (lukasheinrich) [inline .travis.yml]: what version of numpy is needed? Can we encode this in `setup.py`?

Comment (matthewfeickert) [reply .travis.yml]: Will check and try to fix this there.

Comment (matthewfeickert) [reply .travis.yml]: @lukasheinrich Fixed in commit a7d3c3a

File: .travis.yml [hunk 2]
```diff
@@ -4,25 +4,17 @@ python:
...
-  - export PATH="$HOME/miniconda/bin:$PATH"
-  - hash -r
-  - conda config --set always_yes yes --set changeps1 no
-  - conda update -q conda
-  - conda info -a
-  - conda create -q -n test-environment python=$TRAVIS_PYTHON_VERSION pytorch scipy -c pytorch
-  - source activate test-environment
-  - pip install -U --process-dependency-links -e .[develop]
-  - pip install --upgrade pytest
+- sudo apt-get update
```
Comment (lukasheinrich) [inline .travis.yml]: seems like we can remove this

File: .travis.yml [hunk 3]
```diff
@@ -42,25 +34,15 @@ jobs:
     python: '3.6'
     install:
       - sudo apt-get update
```
Comment (lukasheinrich) [inline .travis.yml]: dito

File: .travis.yml (hunk 2 above)
Comment (matthewfeickert) [reply .travis.yml]: Removed in commit 6311608

**Comment by lukasheinrich** [comment]
awesome -- this is a great improvement. thanks a lot @matthewfeickert

**Pull request merged**